src/Neo/SmartContract/JsonSerializer.cs

```diff
@@ -191,8 +191,7 @@ private static StackItem Deserialize(JToken json, ref uint maxStackSize, Referen
                 case JNumber num:
                     {
                         if ((num.Value % 1) != 0) throw new FormatException("Decimal value is not allowed");
-
-                        return (BigInteger)num.Value;
+                        return BigInteger.Parse(num.Value.ToString(), System.Globalization.NumberStyles.Float);
```

Comment:
That should be enough to fit integers that are allowed by our 256-bit VM. I've checked this fix against the neo-go node, and now both nodes are able to handle big numbers correctly.